src/Utilities/Extensions/ISymbolExtensions.cs

```diff
@@ -92,6 +93,24 @@ public static ImmutableArray<IParameterSymbol> GetParameters(this ISymbol symbol
                 _ => ImmutableArray.Create<IParameterSymbol>());
         }
 
+        /// <summary>
+        /// Returns true if the given symbol has required visibility based on options:
+        ///   1. If user has explicitly configured candidate <see cref="SymbolVisibilityGroup"/> in editor config options and
+        ///      given symbol's visibility is one of the candidate visibilites.
+        ///   2. Otherwise, if user has not configured visibility, and given symbol's visibility
+        ///      matches the given default symbol visibility.
+        /// </summary>
+        public static bool MatchesConfiguredVisibility(
+            this ISymbol symbol,
+            AnalyzerOptions options,
+            DiagnosticDescriptor rule,
+            CancellationToken cancellationToken,
+            SymbolVisibilityGroup defaultRequiredVisibility = SymbolVisibilityGroup.Public)
+        {
+            var allowedVisibilities = options.GetSymbolVisibilityGroupOption(rule, defaultRequiredVisibility, cancellationToken);
```

Comment:
why there are 2 implementation of this?

Reply:
I didn't understand, there is just a single implementation here?